src/cljc/schema/spec/collection.cljc

```diff
@@ -23,19 +23,18 @@
       (let [_ (macros/assert! (= 2 (count e)) "remaining can have only one schema.")
             c (spec/sub-checker (second e) params)]
         #?(:clj (fn [^java.util.List res x]
-                  (doseq [i x]
-                    (.add res (c i)))
+                  (reduce (fn [res i] (doto res (.add (c i)))) res x)
                   (then res nil))
            :cljs (fn [res x]
-                   (swap! res into (map c x))
+                   (reduce (fn [res i] (doto res (.push (c i)))) res x)
```

Comment:
I believe that something mutable could be faster here, but I'm again unsure about the use of `reduce`.  Can you share profiling results?